podcasts/End of Year/Stories/DynamicBackgroundView.swift

```diff
@@ -34,13 +30,64 @@ struct DynamicBackgroundView: View {
                 Image("bottom_blob")
                     .resizable()
                     .renderingMode(.template)
-                    .foregroundColor(foregroundColor)
+                    .foregroundColor(model.foregroundColor)
             }
 
         }
     }
 }
 
+@MainActor
+class DynamicBackgroundProvider: ObservableObject {
+    @Published var backgroundColor: Color
+    @Published var foregroundColor: Color
+
+    private let podcast: Podcast?
+
+    init(podcast: Podcast) {
+        backgroundColor = podcast.bgColor().color
+        foregroundColor = ColorManager.lightThemeTintForPodcast(podcast).color
+        self.podcast = podcast
+
+        if !ColorManager.podcastHasBackgroundColor(podcast) {
+            _ = ColorManager.backgroundColorForPodcast(podcast)
+            NotificationCenter.default.addObserver(self, selector: #selector(podcastColorsLoaded(_:)), name: Constants.Notifications.podcastColorsDownloaded, object: nil)
```

Comment:
Since the `podcastColorsDownloaded` notification posts using the Podcast UUID as the object, we can pass it in here to only receive the notification for this podcast, which also means we can reduce the boilerplate checking code needed in `podcastColorsLoaded()`.

```suggestion
            NotificationCenter.default.addObserver(self, selector: #selector(podcastColorsLoaded(_:)), name: Constants.Notifications.podcastColorsDownloaded, object: podcast.uuid)
```

Reply:
Done in d829d221e54c88516e2caecfe5235137941c4650